Pod/Classes/LSClockState.m

```diff
@@ -90,10 +90,11 @@ - (void)addSampleWithOriginMicros:(SInt64)originMicros
     // Note that self.samplesQueue is DISPATCH_QUEUE_SERIAL and it makes thread safe reads/writes.
     dispatch_async(self.samplesQueue, ^{
         // Discard the oldest sample and push the new one.
-        [self.samples removeObjectAtIndex:0];
-        [self.samples
-         addObject:[[LSSyncSample alloc] initWithDelayMicros:latestDelayMicros offsetMicros:latestOffsetMicros]];
-
+        if (self.samples.count > 0) {
```

Comment:
Not sure if I understand the issue - are you saying you'd expect just one value to be cached vs array of values?
I haven't written this code though, trying to minimize diffs